src/spikeinterface/core/analyzer_extension_core.py

```diff
@@ -298,6 +298,17 @@ def _set_params(self, ms_before: float = 1.0, ms_after: float = 2.0, operators=N
 
         waveforms_extension = self.sorting_analyzer.get_extension("waveforms")
         if waveforms_extension is not None:
+
+            ms_before_waveforms = waveforms_extension.nbefore / self.sorting_analyzer.sampling_frequency * 1000
+            ms_after_waveforms = waveforms_extension.nafter / self.sorting_analyzer.sampling_frequency * 1000
+
+            if ms_before != ms_before_waveforms or ms_after != ms_after_waveforms:
+                raise ValueError(
+                    "templates extension parameters are inconsistent with the waveforms extension. \n"
+                    f"waveform extension ms_before={ms_before_waveforms}  ms_after={ms_after_waveforms} \n"
+                    f"templates extension ms_before={ms_before}  ms_after={ms_after}."
+                )
+
```

Comment:
I think I prefered the previous behavior:
  * use parameters when compute directly
  * ignore parameters and hinertis from waveforms extension when this exists.

Reply:
Agree with @samuelgarcia here